Tests/RoslynPluginGeneratorTests/NuGetPackageHandlerTests.cs

```diff
@@ -17,22 +19,211 @@ public class NuGetPackageHandlerTests
+        private const string dependentPackageName = "dependentPackage";
+
+        private const string releaseVersion = "1.0.0";
+        private const string preReleaseVersion = "1.0.0-RC1";
+
         [TestMethod]
-        public void NuGet_TestDependencyResolutionFailure()
+        public void NuGet_TestPackageDownload_Release_Release()
         {
             // Arrange
             string testDir = TestUtils.CreateTestDirectory(this.TestContext);
+            string testDownloadDir = Path.Combine(testDir, "download");
+
+            // Create test NuGet payload and packages
+            string testPackageFile = Path.Combine(testDir, "testPackage.nupkg");
+            string dependentPackageFile = Path.Combine(testDir, "dependentPackage.nupkg");
+
+            ManifestMetadata testMetadata = new ManifestMetadata()
+            {
+                Authors = "Microsoft",
+                Version = releaseVersion,
+                Id = testPackageName,
+                Description = "A description",
+            };
+            buildPackage(testMetadata, testPackageFile);
+
+            List<ManifestDependencySet> dependencies = new List<ManifestDependencySet>()
+            {
+                new ManifestDependencySet()
+                {
+                    Dependencies = new List<ManifestDependency>()
+                    {
+                        new ManifestDependency()
+                        {
+                            Id = testPackageName,
+                            Version = releaseVersion,
+                        }
+                    }
+                }
+            };
+            ManifestMetadata dependentMetadata = new ManifestMetadata()
+            {
+                Authors = "Microsoft",
+                Version = releaseVersion,
+                Id = dependentPackageName,
+                Description = "A description",
+                DependencySets = dependencies,
+            };
+            buildPackage(dependentMetadata, dependentPackageFile);
+
             TestLogger logger = new TestLogger();
             NuGetPackageHandler handler = new NuGetPackageHandler(logger);
 
             // Act
-            // Fetch a package that should fail due to pre-release dependencies
-            IPackage package = handler.FetchPackage(AnalyzerPluginGenerator.NuGetPackageSource, "codeCracker", null, testDir);
+            // Attempt to download a package which is released with a dependency that is released
+            IPackage package = handler.FetchPackage(testDir, dependentPackageName, null, testDownloadDir);
 
             // Assert
-            // No files should have been downloaded
-            Assert.IsTrue(Directory.GetFiles(testDir).Length == 0);
-            Assert.IsNull(package);
+            // Package should have been downloaded
+            AssertPackageDownloaded(testDownloadDir, dependentPackageName);
+            Assert.IsNotNull(package, "Expected a reference to the package to be returned");
```

Comment:
Repeated in every test -> factor out.
Checking the return value first (i.e. before checking the file system) seems more natural.